cupyd/core/communication/connector.py

```diff
@@ -100,6 +105,14 @@ class InterProcessConnector(Connector):
 
     def __init__(self, maxsize: Optional[int] = 0):
         super().__init__(maxsize=maxsize)
+
+        if maxsize and maxsize > SemLock.SEM_VALUE_MAX:
+            logger.warning(
+                f"Limited maxsize for InterProcessConnector (Queue) since is bigger than OS limit: "
+                f"{maxsize} > {SemLock.SEM_VALUE_MAX}"
+            )
+            self._maxsize = SemLock.SEM_VALUE_MAX
+
```

Comment:
this change was done with macOS in mind, but may apply to every other SO (in windows for example, this limit is ridiculously high)